commcare/src/main/java/org/motechproject/commcare/events/FormEventHandler.java

```diff
@@ -0,0 +1,83 @@
+package org.motechproject.commcare.events;
+
+import org.apache.commons.lang.StringUtils;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.motechproject.commcare.domain.FormValueElement;
+import org.motechproject.commcare.domain.FormXml;
+import org.motechproject.commcare.domain.MetadataValue;
+import org.motechproject.commcare.events.constants.EventSubjects;
+import org.motechproject.commcare.service.CommcareFormService;
+import org.motechproject.event.MotechEvent;
+import org.motechproject.event.listener.annotations.MotechListener;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * This class serves as the event handler for the task actions, exposed by the Commcare module.
+ * Respective methods extract the necessary data from the {@link MotechEvent} instance and
+ * pass them to the {@link CommcareFormService} that handles all the operations on Commcare forms.
+ */
+@Component
+public class FormEventHandler {
+
+    private static final String INSTANCE_ID_KEY = "instanceID";
+    private static final String USERNAME_KEY = "username";
+    private static final String TIME_START_KEY = "timeStart";
+    private static final String TIME_END_KEY = "timeEnd";
+
+    private static final String USERNAME_VALUE = "MOTECH";
+
+    private static final DateTimeFormatter COMMCARE_DATETIME_FORMAT = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+
+    @Autowired
+    private CommcareFormService commcareFormService;
+
+    @MotechListener(subjects = EventSubjects.SEND_FORM + ".*")
+    public void sendForm(MotechEvent event) {
+        String configName = EventSubjects.getConfigName(event.getSubject());
+        String xmlns = StringUtils.removeEnd(event.getSubject().replaceFirst(EventSubjects.SEND_FORM + ".", StringUtils.EMPTY), "." + configName);
```

Comment:
I don't get it, is the config name appended at the end after the xmlns? I would expect it the other way around.

Probably a good candidate for a comment at least.